src/test_utils.rs

```diff
@@ -44,14 +43,41 @@ pub fn enable_log(filter: impl Into<EnvFilter>) {
     });
 }
 
+/// Which type of Address do you want? Random may give ipv4 or ipv6
+pub enum AddressType {
+    Random,
+    Ipv4,
+    Ipv6,
+}
+
 /// Returns a local address on a port that is not assigned to another test.
-pub async fn available_addr() -> SocketAddr {
+/// If Random address tye is used, it might be v4, Might be v6. It's random.
+pub async fn available_addr(address_type: &AddressType) -> SocketAddr {
     let socket = create_socket().await;
-    let addr = socket.local_addr().unwrap();
+    let addr = get_address(address_type, &socket);
+
     tracing::debug!(addr = ?addr, "test_util::available_addr");
     addr
 }
 
+fn get_address(address_type: &AddressType, socket: &DualStackLocalSocket) -> SocketAddr {
+    let addr = match address_type {
+        AddressType::Random => {
+            let mut rng = thread_rng();
+            // sometimes give ipv6, sometimes ipv4.
+            match rng.gen_range(0..2) {
+                0 => socket.local_ipv6_addr().unwrap(),
+                1 => socket.local_ipv4_addr().unwrap(),
+                _ => unreachable!(),
+            }
```

Comment:
You can simplify this by using `rand::random`.
```suggestion
            match rand::random() {
                true => socket.local_ipv6_addr().unwrap(),
                false => socket.local_ipv4_addr().unwrap(),
            }
```